File: backend/app/services/stats.py
# @TASK P2-R3-T1 - Platform Stats business logic (aggregate queries)
# @SPEC docs/planning/02-trd.md#platform-stats-api
"""Platform statistics service: aggregate counts for public dashboard.

@TEST tests/api/test_stats.py
"""
import logging

from sqlalchemy import select, func
from sqlalchemy.ext.asyncio import AsyncSession

from app.models.ai_model import AIModel
from app.models.order import Order
from app.models.user import User

logger = logging.getLogger(__name__)


async def get_platform_stats(db: AsyncSession) -> dict[str, int]:
    """Compute platform-wide statistics.

    Queries:
        total_models:   COUNT of AIModel WHERE status = 'active'
        total_bookings: COUNT of Order   WHERE status = 'completed'
        total_brands:   COUNT of User    WHERE role = 'brand' AND is_active = True

    Args:
        db: Async database session.

    Returns:
        Dictionary with total_models, total_bookings, and total_brands.
    """
    # Count active AI models
    models_stmt = select(func.count(AIModel.id)).where(
        AIModel.status == "active"
    )
    models_result = await db.execute(models_stmt)
    total_models = models_result.scalar_one()

    # Count completed orders (bookings)
    bookings_stmt = select(func.count(Order.id)).where(
        Order.status == "completed"
    )
    bookings_result = await db.execute(bookings_stmt)
    total_bookings = bookings_result.scalar_one()

    # Count active brand users
    brands_stmt = select(func.count(User.id)).where(
        User.role == "brand",
        User.is_active == True,  # noqa: E712
    )
    brands_result = await db.execute(brands_stmt)
    total_brands = brands_result.scalar_one()

    return {
        "total_models": total_models,
        "total_bookings": total_bookings,
        "total_brands": total_brands,
    }

File: frontend/src/services/stats.ts
// @TASK P2-S1-T1 - Platform stats service (API client)
// @SPEC specs/screens/home.yaml#data_requirements

import api from './api';
import type { PlatformStats } from '@/types/stats';

// Mock 모드: 백엔드 없이 Mock 데이터 사용
const USE_MOCK = process.env.NEXT_PUBLIC_USE_MOCK === 'true' || true;

// Mock stats data
const MOCK_STATS: PlatformStats = {
  total_models: 12,
  total_bookings: 156,
  total_brands: 48,
};

/**
 * Platform statistics API service
 */
class StatsService {
  /**
   * Get platform-wide statistics
   */
  async getPlatformStats(): Promise<PlatformStats> {
    // Mock 모드
    if (USE_MOCK) {
      await new Promise((r) => setTimeout(r, 200));
      return MOCK_STATS;
    }

    const { data } = await api.get<PlatformStats>('/api/stats');
    return data;
  }
}

export const statsService = new StatsService();
